symbols.js

```diff
@@ -0,0 +1,148 @@
+'use strict'
```

Comment:
@NickNaso can you please explain under what circumstances a user of this repo would need the list of symbols exposed as a JS array? Also, if we do need them, would it be possible to have two arrays? One for symbols from js_native_api.h and one for those from node_api.h?

Reply:
+1. I think explaining why they are there and how they can be used would be good. I also wonder if we need to somehow group them by what is available in each NODE-API version?

Reply:
@gabrielschulhof my idea is to give the possibility to use the module in a build process or using it in other module that needs to know the inclusion path or the list of the symbols exported:

Example of usage of the exported symbols on a build process:
```
node -p "require('./index').symbols.join('\n')" > node_api.def
dlltool -d node_api.def -y libnode.a
```
Example of usage in other JacaScript module:
 ```js
'use strict'
const headers = require('node-api-headers')
// Do something
```
Yes it's possible to provide the list of symbols for:
- `js_native_api.h`
- `node_api.h`

My idea is to change the `symbols.js` in something like this:

```js
'use strict'

const js_native_api_symbols = [
    // Here the list of the symbols for js_native_api.h
]

const node_api = [
     ...js_native_api_symbols,
    // Here the list of symbols for node_api.h
]

module.exports = {
    js_native_api_symblos,
    node_api_symbols
}
```
For me the symbol exported by `js_native_api_symbols` should be considered a subset of those exported by `node_api.h`

Reply:
@mhdawson It's my intentio explain the useage in documentation after we end this first step on implementation. We could group the symbols by Node-API version so in that case we should expose them like:

```js
'use strict'

module.exports = {
    V1: {
        js_native_api_symblos: [
            // List of symbols in js_native_api.h for a specific version
        ],
        node_api_symblos: [
          // List of symbloas in node_api.h for a specific version
        ]
    },
   // ...
}
``` 
I think that list the symblos by version is a better approach.

Reply:
Maybe we could group the symblols considering the experimental cases like below:

```js
'use strict'

module.exports = {
    V1: {
        js_native_api_symblos: [
            // List of symbols in js_native_api.h for a specific version
        ],
        node_api_symblos: [
          // List of symbols in node_api.h for a specific version
        ],
        experimental: [
         // List of symbols for experimental features
       ]
    },
   // ...
}
```

Reply:
I really think we should exclude the experimental ones. I think it will just come back to haunt us if we include them because they change without an associated version number.  In the example above, there may be experimental symbols that were in some Node.js versions that support V1, but not in others so it would not necessarily be correct to included them in the experimental list for V1. As another example an experimental symbol could be experimental during both the timeframe of V1 and V2, so might need to appear in both. It's just too complicated to get right in my opinion....